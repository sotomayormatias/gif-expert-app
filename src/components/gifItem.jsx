import PropTypes from 'prop-types';

export const GifItem = ({ title, src }) => (
    <div className="card">
        <img src={src} alt={title} />
        <p>{ title }</p>
    </div>
)

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired
}