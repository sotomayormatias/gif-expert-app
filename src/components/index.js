export * from './gifGrid';
export * from './gifItem';
export * from './input';