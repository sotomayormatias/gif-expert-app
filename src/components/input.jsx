import PropTypes from "prop-types";
import { useState } from "react"

export const Input = ({ onSearch }) => {
    const [ inputValue, setInputValue ] = useState('');

    const handleChange = ({ target }) => {
        setInputValue(target.value);
    }

    const handleSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim()){
            onSearch(inputValue.trim());
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit} aria-label="form">
            <input
                type='text'
                placeholder='Buscar'
                value={inputValue}
                onChange={handleChange}
            />
        </form>
    );
}

Input.propTypes = {
    onSearch: PropTypes.func.isRequired,
}