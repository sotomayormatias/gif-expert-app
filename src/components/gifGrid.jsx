import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./gifItem";

export const GifGrid = ({category}) => {

    const { gifs, isLoading } = useFetchGifs(category);

    return (
        <>
            <h2>{category}</h2>
            {
                isLoading 
                    ? <h2>Cargando...</h2>
                    : <div className='card-grid'>
                        {gifs.map(gif => (
                            <GifItem key={gif.id} {...gif} />
                        ))}
                    </div>
            }
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}