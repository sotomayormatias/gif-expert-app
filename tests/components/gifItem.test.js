import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/gifItem";

describe('<GifItem /> test', () => {
    const titulo = 'titulo';
    const source = 'https://source.com/example.jpg';

    test('sholud', () => {
        const {container} = render(<GifItem title={titulo} src={source} />);
        expect(container).toMatchSnapshot();
    });

    test('should get img with title and source', () => {
        render(<GifItem title={titulo} src={source} />);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(source);
        expect(alt).toBe(titulo);
    });

    test('should get title', () => {
        render(<GifItem title={titulo} src={source} />);
        expect(screen.getByText(titulo)).toBeTruthy();
    });
})