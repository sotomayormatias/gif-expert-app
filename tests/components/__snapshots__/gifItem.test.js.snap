// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> test sholud 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://source.com/example.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
